### PR TITLE
Add authorization router. Add tests.

### PR DESCRIPTION
**Pull request by Kun-Kun**
Added router, for now without a factory, but I will implement it after merge all rourers to master branch.

### CODE REVIEW

File: softgroup/services/authorization/authorization-impl/src/main/java/com/softgroup/authorization/impl/handler/LoginRequestHandler.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import com.softgroup.authorization.api.router.AuthorizationRequestHandler;
+import com.softgroup.common.protocol.*;
+import com.softgroup.common.router.api.AbstractRequestHandler;
+import org.springframework.stereotype.Component;
+
+/**
+ * Created by user on 24.02.2017.
+ */
+@Component
+public class LoginRequestHandler extends AbstractRequestHandler<LoginRequest,LoginResponse>  implements AuthorizationRequestHandler{
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-impl/src/main/java/com/softgroup/authorization/impl/handler/LoginRequestHandler.java]: @Kun-Kun Where is handle method?? How is it possible that you have handler without method to handle message O_O ?

File: softgroup/services/authorization/authorization-impl/src/main/java/com/softgroup/authorization/impl/handler/RegisterRequestHandler.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import com.softgroup.common.protocol.Request;
+import com.softgroup.common.protocol.Response;
+import com.softgroup.common.router.api.AbstractRequestHandler;
+import org.springframework.stereotype.Component;
+
+/**
+ * Created by user on 24.02.2017.
+ */
+@Component
+public class RegisterRequestHandler extends AbstractRequestHandler<RegisterRequest,RegisterResponse> implements AuthorizationRequestHandler {
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-impl/src/main/java/com/softgroup/authorization/impl/handler/RegisterRequestHandler.java]: @Kun-Kun Same.

File: softgroup/services/authorization/authorization-impl/src/main/java/com/softgroup/authorization/impl/handler/SmsConfirmRequestHandler.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import com.softgroup.common.protocol.Request;
+import com.softgroup.common.protocol.Response;
+import com.softgroup.common.router.api.AbstractRequestHandler;
+import org.springframework.stereotype.Component;
+
+/**
+ * Created by user on 25.02.2017.
+ */
+    @Component
+    public class SmsConfirmRequestHandler extends AbstractRequestHandler<SmsConfirmRequest,SmsConfirmResponse> implements AuthorizationRequestHandler {
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-impl/src/main/java/com/softgroup/authorization/impl/handler/SmsConfirmRequestHandler.java]: @Kun-Kun Same

File: softgroup/services/authorization/authorization-impl/src/main/java/com/softgroup/authorization/impl/router/AuthorizationRouterAppCfg.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.softgroup.authorization.impl.router;
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-impl/src/main/java/com/softgroup/authorization/impl/router/AuthorizationRouterAppCfg.java]: @Kun-Kun Why router? This configuration is not only for router.

File: softgroup/services/authorization/authorization-impl/src/main/java/com/softgroup/authorization/impl/handler/LoginRequestHandler.java (hunk 1 above)
Comment (Kun-Kun) [inline softgroup/services/authorization/authorization-impl/src/main/java/com/softgroup/authorization/impl/handler/LoginRequestHandler.java]: fixed

File: softgroup/services/authorization/authorization-impl/src/main/java/com/softgroup/authorization/impl/handler/RegisterRequestHandler.java (hunk 2 above)
Comment (Kun-Kun) [inline softgroup/services/authorization/authorization-impl/src/main/java/com/softgroup/authorization/impl/handler/RegisterRequestHandler.java]: fixed

File: softgroup/services/authorization/authorization-impl/src/main/java/com/softgroup/authorization/impl/handler/SmsConfirmRequestHandler.java (hunk 3 above)
Comment (Kun-Kun) [inline softgroup/services/authorization/authorization-impl/src/main/java/com/softgroup/authorization/impl/handler/SmsConfirmRequestHandler.java]: fixed

File: softgroup/services/authorization/authorization-impl/src/main/java/com/softgroup/authorization/impl/router/AuthorizationRouterAppCfg.java (hunk 4 above)
Comment (Kun-Kun) [inline softgroup/services/authorization/authorization-impl/src/main/java/com/softgroup/authorization/impl/router/AuthorizationRouterAppCfg.java]: You mean to make a configuration in a separate package?

**Comment by Dominys** [comment]
@Kun-Kun Don't merge master into branch! Use re-base.

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 5]
```diff
@@ -1,16 +1,32 @@
...
+		Request<RQ> typedRequest = toType(msg);
+		Response<RS> response = processRequest(typedRequest);
+		return response;
+	}
+
+
+	private  Request<RQ> toType(Request<?> msg ){
+		Object dataObject = msg.getData();
+		Map<String,Object> map =  dataMapper.convertToMap(dataObject);
+		RQ requestData = dataMapper.convert(map, new TypeReference<RQ>() {});
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java]: @Kun-Kun Why two conversions ?

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 6]
```diff
@@ -1,16 +1,32 @@
...
+
+	@Autowired
+	DataMapper dataMapper;
 
 	@Override
-	public Response<R> handle(Request<?> msg) {
-		return null;
+	public Response<RS> handle(Request<?> msg) {
+		Request<RQ> typedRequest = toType(msg);
+		Response<RS> response = processRequest(typedRequest);
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java]: @Kun-Kun Return immediately

File: softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/message/RegisterResponse.java [hunk 7]
```diff
@@ -8,4 +8,32 @@
  */
 public class RegisterResponse implements ResponseData {
 	private static final long serialVersionUID = -5146888202653750948L;
+
+	private String registrationRequestUuid;
+	private String registrationTimeoutSec;
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/message/RegisterResponse.java]: @Kun-Kun Integer

**Pull request merged**